apps/nova/src/pages/UpcomingLessons.tsx

```diff
@@ -21,10 +55,35 @@ const UpcomingLessons: React.FC = () => {
     canceled: true,
   });
 
+  // useMutation hooks
+  const { mutate: createRating } = useCreateRatingTutor({
+    onSuccess: () => setRateDialogInfo((prev) => ({ ...prev, enabled: false })),
+    onError: () => console.log("createRating failed!"),
+  });
+
+  const { mutate: editRating } = useEditRatingTutor({
+    onSuccess: () => setRateDialogInfo((prev) => ({ ...prev, enabled: false })),
+    onError: () => console.log("editRating failed!"),
+  });
+
+  // useEffect hooks
   useEffect(() => {
     if (!user) return navigate(Route.Root);
   }, [navigate, user]);
 
+  useEffect(() => {
+    if (user?.role !== "student") return;
+    const { tutorId, lessonId } = getLessonQuery(params);
+    setRateDialogInfo((prev) => ({ ...prev, tutorId, lessonId }));
+  }, [params, user?.role]);
```

Comment:
It is a good practice to clean the url search params after extracing this value. This way if the user refersed the page he will not get the same dialog again.

```suggestion
  useEffect(() => {
    if (user?.role !== "student") return;
    const { tutorId, lessonId } = getLessonQuery(params);
    setRateDialogInfo((prev) => ({ ...prev, tutorId, lessonId }));
    // reset the url as the data is already captured in the local state. 
    setSearchParams({}); 
  }, [params, user?.role]);
```